docs/api/dns.md

```diff
@@ -8,34 +8,106 @@ console.log(addrs);
+- `Bun.connect`
+- `node:net`
+- `node:tls`
+
+### When should I prefetch a DNS entry?
+
+Web browsers expose [`<link rel="dns-prefetch">`](https://developer.mozilla.org/en-US/docs/Web/Performance/dns-prefetch) to allow developers to prefetch DNS entries. This is useful when you know you'll need to connect to a host in the near future and want to avoid the initial DNS lookup.
+
+In Bun, you can use the `dns.prefetch` API to achieve the same effect. 
+
 ```ts
-import { dns } from "bun";
-const [{ address }] = await dns.lookup("example.com");
-console.log(address); // "93.184.216.34"
+import {dns} from "bun";
+
+dns.prefetch("my.database-host.com", 5432);
 ```
-If you need to limit IP addresses to either IPv4 or IPv6, you can specify the `family` as an option.
+
+An example where you might want to use this is a database driver. When your application first starts up, you can prefetch the DNS entry for the database host so that by the time it finishes loading everything, the DNS query to resolve the database host may already be completed.
+
+### `dns.prefetch`
+
+{% callout %}
+**🚧** — This API is experimental and may change in the future.
+{% /callout %}
+
+To prefetch a DNS entry, you can use the `dns.prefetch` API. This API is useful when you know you'll need to connect to a host in the soon and want to avoid the initial DNS lookup.
+
 ```ts
-import { dns } from "bun";
-const [{ address }] = await dns.lookup("example.com", { family: 6 });
-console.log(address); // "2606:2800:220:1:248:1893:25c8:1946"
+dns.prefetch(hostname: string, port: number): void;
 ```
-Bun supports three backends for DNS resolution:
-- `c-ares` - This is the default on Linux, and it uses the [c-ares](https://c-ares.org/) library to perform DNS resolution.
-- `system` - Uses the system's non-blocking DNS resolver, if available. Otherwise, falls back to `getaddrinfo`. This is the default on macOS, and the same as `getaddrinfo` on Linux.
-- `getaddrinfo` - Uses the POSIX standard `getaddrinfo` function, which may cause performance issues under concurrent load.
 
-You can choose a particular backend by specifying `backend` as an option.
+Here's an example:
+
 ```ts
-import { dns } from "bun";
-const [{ address, ttl }] = await dns.lookup("example.com", {
-  backend: "c-ares"
-});
-console.log(address); // "93.184.216.34"
-console.log(ttl); // 21237
+import {dns} from "bun";
+
+dns.prefetch("bun.sh", 443);
+//
+// ... sometime later ...
+await fetch("https://bun.sh");
 ```
-Note: the `ttl` property is only accurate when the `backend` is c-ares. Otherwise, `ttl` will be `0`.
-This was added in Bun v0.5.0. -->
+
+### `dns.getCacheStats()`
+
+{% callout %}
+**🚧** — This API is experimental and may change in the future.
+{% /callout %}
+
+To get the current cache stats, you can use the `dns.getCacheStats` API. 
+
+This API returns an object with the following properties:
+
+```ts
+{
+  // Cache hits 
+  cacheHitsCompleted: number;
+  cacheHitsInflight: number;
+  cacheMisses: number;
```

Comment:
would be useful to see these values per hostname as well to see where lookup activity is concentrated. could generate some useful graphs from this too.